azext_iot/commands.py

```diff
@@ -50,11 +50,54 @@ def load_command_table(self, _):
         )
         cmd_group.command("import", "iot_device_import")
         cmd_group.command("export", "iot_device_export")
-        cmd_group.command("add-children", "iot_device_children_add")
-        cmd_group.command("remove-children", "iot_device_children_remove")
-        cmd_group.command("list-children", "iot_device_children_list")
-        cmd_group.command("get-parent", "iot_device_get_parent")
-        cmd_group.command("set-parent", "iot_device_set_parent")
+        cmd_group.command(
+            "add-children",
+            "iot_device_children_add",
+            deprecate_info=self.deprecate(
+                redirect="az iot hub device-identity children add"
+            ),
+        )
+        cmd_group.command(
+            "remove-children",
+            "iot_device_children_remove",
+            deprecate_info=self.deprecate(
+                redirect="az iot hub device-identity children remove"
+            ),
+        )
+        cmd_group.command(
+            "list-children",
+            "iot_device_children_list",
+            deprecate_info=self.deprecate(
+                redirect="az iot hub device-identity children list"
+            ),
+        )
+        cmd_group.command(
+            "get-parent",
+            "iot_device_get_parent",
+            deprecate_info=self.deprecate(
+                redirect="az iot hub device-identity parent get"
+            ),
+        )
+        cmd_group.command(
+            "set-parent",
+            "iot_device_set_parent",
+            deprecate_info=self.deprecate(
+                redirect="az iot hub device-identity parent set"
+            ),
+        )
+
+    with self.command_group(
+        "iot hub device-identity children", command_type=iothub_ops
+    ) as cmd_group:
+        cmd_group.show_command("add", "iot_device_children_add")
+        cmd_group.show_command("remove", "iot_device_children_remove")
+        cmd_group.show_command("list", "iot_device_children_list")
+
+    with self.command_group(
+        "iot hub device-identity parent", command_type=iothub_ops
+    ) as cmd_group:
+        cmd_group.show_command("get", "iot_device_get_parent")
```

Comment:
`get` and `set` sound like a nice pair, but overall we want the single resource retrievals to be `show` to align with Az CLI verbs.

Reply:
the `add` and `remove` above are OK because that is adding or removing entities from a collection.